mkwindows.sh

```diff
@@ -0,0 +1,228 @@
+    mkdir ../webp/lib
+    i686-w64-mingw32-gcc -shared -fPIC -DPIC  -Wl,--whole-archive src/dec/.libs/libwebpdecode.a src/dsp/.libs/libwebpdsp.a src/enc/.libs/libwebpencode.a src/utils/.libs/libwebputils.a -Wl,--no-whole-archive -lm -O2 -pthread -o ../webp/lib/libwebp.dll
+cd ../..
+if ! [ -r ${WEBP_DIR}/include/webp/decode.h -a -r ${WEBP_DIR}/lib/libwebp.a ] ; then
+    # I expect that cross-compiling webp is going to be very fragile,
+    # so print a nice error in case this happens.
+    echo "Cross-compilation apparently failed:"
+    echo "Some files in ${WEBP_DIR} are missing."
+    echo "Please submit a bugreport."
+fi
+
+
+# === Build auto-gen files ===
+
+# Create and backup all auto/ files
+echo "===== 2: Create auto/ files"
+make tgl/Makefile
+make -C tgl -j4 auto/auto-autocomplete.c auto/auto.c auto/auto-fetch.c auto/auto-fetch-ds.c auto/auto-free-ds.c
+make -C tgl -j4 auto/auto-print-ds.c auto/auto-skip.c auto/auto-store.c auto/auto-store-ds.c auto/auto-types.c
+make bin
+tar cf bin/tgl-auto-code.tar tgl/auto/
+
+
+# === Cross-compilation itself ===
+
+# Clean tgl only (do not touch bin/ with the auto-gen files!)
+echo "===== 3: Cleanup tgl"
+make -C tgl clean
+
+# Reconfigure
+echo "===== 4: Configure for cross-compilation"
+# Sadly, the library paths must be resolved *now* already.
+# AND, they must be absolute.  Sigh.
+export WIN32_GTK_DEV_DIR=`realpath win32/gtk_2_0-2.16`
+export WIN32_PIDGIN_DIR=`realpath win32/pidgin-2.12.0`
+export WIN32_WEBP_DIR="${WEBP_DIR_FULL}"
+./configure -q --host x86_64-linux-gnu --target i686-w64-mingw32 \
+    --includedir="/usr/share/mingw-w64/include" \
+    --oldincludedir="/usr/share/mingw-w64/include" \
+    --libdir="/usr/i686-w64-mingw32" \
+    --libexecdir="/usr/i686-w64-mingw32" \
+    --with-zlib="/usr/i686-w64-mingw32" \
+    CC=i686-w64-mingw32-gcc CFLAGS="-g -O2" \
+    PURPLE_CFLAGS="\${WIN32_INC}" \
+    PURPLE_LIBS="-lpurple -lglib-2.0" \
+    LDFLAGS="-Wl,--export-all-symbols -L${WIN32_GTK_DEV_DIR}/lib -L${WIN32_PIDGIN_DIR}-win32bin -L${WIN32_WEBP_DIR}/lib" \
+    LIBS="-lssp -lintl -lz -lgcrypt -lws2_32"
+cd tgl
+./configure -q --host x86_64-linux-gnu --target i686-w64-mingw32 \
+    --disable-openssl --disable-extf \
+    --includedir="/usr/share/mingw-w64/include" \
+    --oldincludedir="/usr/share/mingw-w64/include" \
+    --libdir="/usr/i686-w64-mingw32" \
+    --libexecdir="/usr/i686-w64-mingw32" \
+    --with-zlib="/usr/i686-w64-mingw32" \
+    CC=i686-w64-mingw32-gcc CFLAGS="-g -O2" LDFLAGS="-lssp"
+cd ..
+
+# Pretend we're building up the preliminaries for auto/
+echo "===== 5: Compile tgl, pre-'generate' files"
+# Need to override LOCAL_LDFLAGS to remove '-rshared'.
+echo "    In case this fails mysteriously with some 'file stubs-32.h not found':"
+echo "    You're probably missing libc6-dev-i386 or some other dev files."
+make -C tgl -j4 LOCAL_LDFLAGS="-lz -lgcrypt -ggdb" bin/generate
+
+# Surprise!  Don't actually call bin/generate to create the autogen files,
```

Comment:
So instead of moving around the autogen files, it should move around `bin/generate`?  Hmm.  Actually, that seems much nicer, and gets rid of those ugly `touch` hacks a few lines down.  Thanks!